### PR TITLE
Implement 'hasOwn' routine of Object object

### PR DESCRIPTION
**Pull request by repasics**
JerryScript-DCO-1.0-Signed-off-by: Csaba Repasi repasics@inf.u-szeged.hu

### CODE REVIEW

File: tests/jerry/es.next/object-hasown.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+assert(Object.hasOwn(proxy1, "foo") == true)
+assert(Object.hasOwn(proxy1, "bar") == false)
+
+assert(Object.hasOwn(proxy2, "foo") == true)
+assert(Object.hasOwn(proxy2, "bar") == true)
+
+assert(Object.hasOwn(proxy3, "foo") == false)
+assert(Object.hasOwn(proxy3, "bar") == false)
+
```
Comment (rerobika) [inline tests/jerry/es.next/object-hasown.js:45]: please add test for Object.hasOwn.call(undefined), and `Object.hasOwn.call({}, {get toString() { throw new Error()}})`

**Review by rerobika — CHANGES_REQUESTED** [review]

File: jerry-core/api/jerryscript.c [hunk 2]
```diff
@@ -3249,27 +3249,11 @@ jerry_has_own_property (const jerry_value_t obj_val, /**< object value */
...
-#if JERRY_BUILTIN_TYPEDARRAY
-  return jerry_return (ecma_op_ordinary_object_has_own_property (obj_p, prop_name_p));
-#else /* !JERRY_BUILTIN_TYPEDARRAY */
-  return ecma_op_ordinary_object_has_own_property (obj_p, prop_name_p);
-#endif /* JERRY_BUILTIN_TYPEDARRAY */
+#if JERRY_BUILTIN_PROXY || JERRY_BUILTIN_TYPEDARRAY
+  return jerry_return (ecma_op_object_has_own_property (obj_p, prop_name_p));
+#else /* !JERRY_BUILTIN_PROXY && !JERRY_BUILTIN_TYPEDARRAY */
+  return ecma_op_object_has_own_property (obj_p, prop_name_p);
+#endif /* JERRY_BUILTIN_PROXY || JERRY_BUILTIN_TYPEDARRAY */
```
Comment (galpeter) [inline jerry-core/api/jerryscript.c:3256]: AFAIK we could simplify this to:
```c
return jerry_return(ecma_op_object_has_own_property(...));
```
I don't think the skipping of `jerry_return` call worth the extra #if-def guard.

**Comment by zherczeg** [comment]
It does not fix any test262 tests, is it a new standard?

**Comment by rerobika** [comment]
> It does not fix any test262 tests, is it a new standard?

We need to update the test262 runner to the latest master. cc @ossy-szeged

File: tests/test262-esnext-excludelist.xml [hunk 3]
```diff
@@ -4450,64 +4450,6 @@
   <!-- ES2022: Object hasOwn
        https://github.com/tc39/proposal-accessible-object-hasownproperty
   -->
-  <test id="built-ins/Object/hasOwn/descriptor.js"><reason></reason></test>
```
Comment (rerobika) [inline tests/test262-esnext-excludelist.xml:4453]: Please remove the comment around

File: jerry-core/ecma/operations/ecma-objects.c [hunk 4]
```diff
@@ -3401,6 +3401,35 @@ ecma_op_ordinary_object_has_own_property (ecma_object_t *object_p, /**< the obje
   return ecma_make_boolean_value (ECMA_PROPERTY_IS_FOUND (property));
 } /* ecma_op_ordinary_object_has_own_property */
 
+/**
+ * Checks whether an object (excluding prototypes) has a named property. Handles proxy objects too.
+ *
+ * @return true - if property is found
+ *         false - otherwise
+ */
+extern inline ecma_value_t
```
[the remark this answers is not in the document]
Comment (rerobika) [reply jerry-core/ecma/operations/ecma-objects.c:3413]: This request is still valid

**Review by rerobika — CHANGES_REQUESTED** [review]

File: jerry-core/ecma/operations/ecma-objects.c [hunk 5]
```diff
@@ -3404,6 +3404,35 @@ ecma_op_ordinary_object_has_own_property (ecma_object_t *object_p, /**< the obje
   return ecma_make_boolean_value (ECMA_PROPERTY_IS_FOUND (property));
 } /* ecma_op_ordinary_object_has_own_property */
 
+/**
+ * Checks whether an object (excluding prototypes) has a named property. Handles proxy objects too.
+ *
+ * @return true - if property is found
+ *         false - otherwise
+ */
+extern ecma_value_t
```
Comment (rerobika) [inline jerry-core/ecma/operations/ecma-objects.c:3413]: I think we can remove the extern keyword

**Review by rerobika — APPROVED** [review]

**Review by ossy-szeged — APPROVED** [review]

**Pull request merged**